pkg/spdk/replica.go

```diff
@@ -1071,6 +1071,51 @@ func (r *Replica) SnapshotRevert(spdkClient *spdkclient.Client, snapshotName str
 	return ServiceReplicaToProtoReplica(r), nil
 }
 
+// SnapshotPurge asks the replica to delete all system created snapshots
+func (r *Replica) SnapshotPurge(spdkClient *spdkclient.Client) (err error) {
+	updateRequired := false
+
+	r.Lock()
+	defer func() {
+		r.Unlock()
+
+		if updateRequired {
+			r.UpdateCh <- nil
+		}
+	}()
+
+	defer func() {
+		if err != nil && r.State != types.InstanceStateError {
+			r.State = types.InstanceStateError
+			updateRequired = true
+		}
+	}()
+
+	if r.ChainLength < 2 {
+		return fmt.Errorf("invalid chain length %d for replica snapshot purge", r.ChainLength)
+	}
+
+	// delete all non-user-created snapshots
+	for snapshotLvolName, snapSvcLvol := range r.SnapshotLvolMap {
+		if snapSvcLvol.UserCreated {
+			continue
+		}
```

Comment:
We have a [bug](https://github.com/longhorn/longhorn/issues/9054#issuecomment-2245049025) in v1.6.2.
The `userCreated` of snapshots for v2 volumes created in v1.6.2 are set to `false`.
The purge will lead to these snapshots being deleted.
We can add this in the `important note`? Is it OK? cc @shuo-wu @DamiaSan @innobead

Reply:
YES. I am fine with it.